app/cdap/components/DataPrep/Directives/Calculate/index.js

```diff
@@ -516,19 +553,28 @@ export default class Calculate extends Component {
   }
 
   parseColumns = () => {
-    const columns = typeof this.props.column === 'string' ? [this.props.column] : this.props.column;
-    const columnTypes = [...new Set(columns.map(col => DataPrepStore.getState().dataprep.typesCheck[col]))];
-    const crossColumn = columns.length === 2;
-    const multiColumn = columns.length > 2;
-    const areColsMixedTypes = columnTypes.length > 1;
+    let columns = typeof this.props.column === 'string' ? [this.props.column] : this.props.column;
+    let columnTypes = [...new Set(columns.map(col => DataPrepStore.getState().dataprep.typesCheck[col]))];
+
+    // We replace 'decimal' with 'bigdecimal' to be compatible with the NUMBER_TYPES list
```

Comment:
Are `decimal` and `bigdecimal` the same?

Reply:
Yes -- BigDecimal is the name of the Java datatype, while Decimal is the name which the Wrangler frontend assigns to columns of that type. 

That led to the issue we're fixing here, where the UI wasn't recognizing that columns containing BigDecimal could have calculation operations performed on them.